### PR TITLE
fix: Change parsing proto to json

### PR DESCRIPTION
**Pull request by inchori**
First of all, I just changed parsing method only in ```createDeal``` using ```jsonpb```.
If my work is correct, I will change method in ```sellData```.

### CODE REVIEW

File: x/market/client/cli/txDeal.go [hunk 1]
```diff
@@ -100,48 +106,23 @@ func CmdDeactivateDeal() *cobra.Command {
...
-	}
-
-	msg := types.NewMsgCreateDeal(
-		deal.DataSchema,
-		&budget,
-		deal.MaxNumData,
-		deal.TrustedDataValidators,
-		clientCtx.GetFromAddress().String(),
-	)
+func NewBuildCreateDealMsg(txf tx.Factory, fs *flag.FlagSet) (tx.Factory, types.MsgCreateDeal, error) {
```
Comment (gyuguen) [inline x/market/client/cli/txDeal.go:109]: Is there a reason to use it like this because it seems like there is no need to return `tx.Factory`?

Comment (inchori) [reply x/market/client/cli/txDeal.go:109]: Oh yes, in this function the ```tx.Factory``` seems to be not needed. 
By the way, there have some problem in unmarshaling the create deal json file. So I drafted this PR.

File: x/market/client/cli/txDeal_test.go [hunk 2]
```diff
@@ -1,26 +1,16 @@
...
+func TestParseSellDataFlags(t *testing.T) {
+	flags := CmdSellData().Flags()
+	err := flags.Set(DataVerificationCertificateFile, "./testdata/data_certificate_file.json")
 
-// TODO: Test Client Command-Line MsgSellData
-func (suite *txTestSuite) TestCmdSellData() {
+	require.NoError(t, err)
 
+	_, err = parseSellDataFlags(flags)
+	require.NoError(t, err)
```
Comment (hansol-medi) [inline x/market/client/cli/txDeal_test.go:15]: Why don't we assert that parsing results are the same to the expected values?

File: x/market/client/cli/txDeal.go [hunk 3]
```diff
@@ -150,57 +150,41 @@ func NewSellDataMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet)
...
-		return txf, nil, err
+		DealId:               sellData.UnsignedCert.DealId,
+		DataHash:             sellData.UnsignedCert.DataHash,
+		EncryptedDataUrl:     sellData.UnsignedCert.EncryptedDataUrl,
+		DataValidatorAddress: sellData.UnsignedCert.DataValidatorAddress,
+		RequesterAddress:     sellData.UnsignedCert.RequesterAddress,
 	}
 
 	signed := types.DataValidationCertificate{
 		UnsignedCert: &unSigned,
```
Comment (gyuguen) [inline x/market/client/cli/txDeal.go:162]:
```suggestion
		UnsignedCert: sellData.UnsignedCert,
```
I think you can use this as it is.

File: x/market/client/cli/txDeal.go [hunk 4]
```diff
@@ -150,57 +150,41 @@ func NewSellDataMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet)
...
-		Signature:    signatureBytes,
+		Signature:    sellData.Signature,
 	}
 
 	msg := types.NewMsgSellData(signed, clientCtx.GetFromAddress().String())
```
Comment (gyuguen) [inline x/market/client/cli/txDeal.go:166]:
```suggestion
	msg := types.NewMsgSellData(sellData, clientCtx.GetFromAddress().String())
```

I think you can use sellData s it is above.
And you might want to change the name of this.

File: x/market/client/cli/txDeal.go (hunk 3 above)
Comment (gyuguen) [reply x/market/client/cli/txDeal.go:162]: Ah, Please ignore this comment.

File: x/market/client/cli/txDeal.go (hunk 4 above)
Comment (inchori) [reply x/market/client/cli/txDeal.go:166]: It seems to better. I'll change it.

File: x/market/client/cli/txDeal_test.go (hunk 2 above)
Comment (inchori) [reply x/market/client/cli/txDeal_test.go:15]: Oh yes, I'll add it.

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Review by gyuguen — APPROVED** [review]
LGTM 👍

**Pull request merged**